subsys/sal/sid_pal/src/sid_mfg_hex_v7.c

```diff
@@ -0,0 +1,174 @@
+    {SID_PAL_MFG_STORE_MAN_PUB_ED25519_SIGNATURE,    SID_PAL_MFG_STORE_MAN_PUB_ED25519_SIGNATURE_SIZE,    SID_PAL_MFG_STORE_OFFSET_MAN_PUB_ED25519_SIGNATURE},
+    {SID_PAL_MFG_STORE_MAN_ED25519_SERIAL,           SID_PAL_MFG_STORE_MAN_ED25519_SERIAL_SIZE,           SID_PAL_MFG_STORE_OFFSET_MAN_ED25519_SERIAL},
+    {SID_PAL_MFG_STORE_MAN_PUB_P256R1,               SID_PAL_MFG_STORE_MAN_PUB_P256R1_SIZE,               SID_PAL_MFG_STORE_OFFSET_MAN_PUB_P256R1},
+    {SID_PAL_MFG_STORE_MAN_PUB_P256R1_SIGNATURE,     SID_PAL_MFG_STORE_MAN_PUB_P256R1_SIGNATURE_SIZE,     SID_PAL_MFG_STORE_OFFSET_MAN_PUB_P256R1_SIGNATURE},
+    {SID_PAL_MFG_STORE_MAN_P256R1_SERIAL,            SID_PAL_MFG_STORE_MAN_P256R1_SERIAL_SIZE,            SID_PAL_MFG_STORE_OFFSET_MAN_P256R1_SERIAL},
+    {SID_PAL_MFG_STORE_SW_PUB_ED25519,               SID_PAL_MFG_STORE_SW_PUB_ED25519_SIZE,               SID_PAL_MFG_STORE_OFFSET_SW_PUB_ED25519},
+    {SID_PAL_MFG_STORE_SW_PUB_ED25519_SIGNATURE,     SID_PAL_MFG_STORE_SW_PUB_ED25519_SIGNATURE_SIZE,     SID_PAL_MFG_STORE_OFFSET_SW_PUB_ED25519_SIGNATURE},
+    {SID_PAL_MFG_STORE_SW_ED25519_SERIAL,            SID_PAL_MFG_STORE_SW_ED25519_SERIAL_SIZE,            SID_PAL_MFG_STORE_OFFSET_SW_ED25519_SERIAL},
+    {SID_PAL_MFG_STORE_SW_PUB_P256R1,                SID_PAL_MFG_STORE_SW_PUB_P256R1_SIZE,                SID_PAL_MFG_STORE_OFFSET_SW_PUB_P256R1},
+    {SID_PAL_MFG_STORE_SW_PUB_P256R1_SIGNATURE,      SID_PAL_MFG_STORE_SW_PUB_P256R1_SIGNATURE_SIZE,      SID_PAL_MFG_STORE_OFFSET_SW_PUB_P256R1_SIGNATURE},
+    {SID_PAL_MFG_STORE_SW_P256R1_SERIAL,             SID_PAL_MFG_STORE_SW_P256R1_SERIAL_SIZE,             SID_PAL_MFG_STORE_OFFSET_SW_P256R1_SERIAL},
+    {SID_PAL_MFG_STORE_AMZN_PUB_ED25519,             SID_PAL_MFG_STORE_AMZN_PUB_ED25519_SIZE,             SID_PAL_MFG_STORE_OFFSET_AMZN_PUB_ED25519},
+    {SID_PAL_MFG_STORE_AMZN_PUB_P256R1,              SID_PAL_MFG_STORE_AMZN_PUB_P256R1_SIZE,              SID_PAL_MFG_STORE_OFFSET_AMZN_PUB_P256R1},
+};
+// clang-format on
+
+int parse_mfg_const_offsets(tlv_ctx *tlv, struct mfg_header *mfg_header)
+{
+	if (tlv->end_offset <= tlv->start_offset) {
+		return -EINVAL;
+	}
+
+	uint32_t size = tlv->end_offset - tlv->start_offset;
+
+	tlv_read_start_marker(tlv, (uint8_t *)mfg_header, sizeof(struct mfg_header));
+	uint8_t *ram_tlv_data = sid_hal_malloc(size);
+	if (ram_tlv_data == NULL) {
+		return -ENOMEM;
+	}
+	memset(ram_tlv_data, 0xff, size);
+
+	tlv_ctx ram_tlv = { .start_offset = 0,
+			    .end_offset = size,
+			    .tlv_storage_start_marker_size = tlv->tlv_storage_start_marker_size,
+			    .storage = { .ctx = ram_tlv_data,
+					 .read = tlv_storage_ram_read,
+					 .write = tlv_storage_ram_write } };
+
+	for (int i = 0; i < ARRAY_SIZE(sid_pal_mfg_store_app_value_to_offset_table); i++) {
+		struct sid_pal_mfg_store_value_to_address_offset element =
+			sid_pal_mfg_store_app_value_to_offset_table[i];
+		uint8_t *payload_buffer = sid_hal_malloc(element.size);
```

Comment:
Is it really necessary to allocate (and later free) memory for each element?
I think that code will be faster and safe when you just use static ram buffer.